ql/src/java/org/apache/hadoop/hive/ql/plan/ConditionalResolverMergeFiles.java

```diff
@@ -271,6 +300,12 @@ private void generateActualTasks(HiveConf conf, List<Task<?>> resTsks,
         Utilities.FILE_OP_LOGGER.debug("merge resolver will merge " + status[i].getPath());
         work.resolveDynamicPartitionStoredAsSubDirsMerge(conf, status[i].getPath(), tblDesc,
             aliases, pDesc);
+        // Do not add input file since its already added when the manifest file is present.
+        if (!manifestFilePresent) {
+          List<Path> inputPaths = work.getPaths();
+          inputPaths.add(status[i].getPath());
```

Comment:
should this be done inside the loop? work.setInputPaths keeps changing...

Reply:
Added `toMerge` to keep track of what needs to be changed. And do final update on the `work` object.